File: my-folder/0209-minimum-size-subarray-sum/solution.java
class Solution {
    public int minSubArrayLen(int s, int[] nums) {
        int n = nums.length;
        int left = 0;
        int right = 0;
        int sum = 0;
        int minLen = n + 1;
        while(right<n){
            sum += nums[right];
            while(sum>=s){
                minLen = Math.min(minLen, right - left + 1);
                sum -= nums[left++];
            }
            right++;
        }
        return minLen == n+1? 0: minLen;
    }
}

File: my-folder/0209-minimum-size-subarray-sum/solution.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        """
        :type target: int
        :type nums: List[int]
        :rtype: int
        """
        min_size = len(nums)+1
        start = 0
        end = 0
        s = 0
        while end<len(nums):
            s+=nums[end]
            if s>=target:
                while s-nums[start]>=target:
                    s -= nums[start]
                    start+=1
                min_size = min(min_size, end-start+1)
            end+=1
        return 0 if min_size==len(nums)+1 else min_size
            
